Initial PR (#1)
osuosl-cookbooks/osl-resources: documentation/osl_authorized_keys.md

# osl_authorized_keys

## Actions

> - `Add`: Adds a key from selected file.
> - `Remove`: Removes a key from selected file.

Reviewer: Please make the actions symbols:
```suggestion
- `:add`: Adds a key from selected file.
- `:remove`: Removes a key from selected file.
```

## Properties

| Property     | Type   | Default          | Required | Description                                                 |
|--------------|--------|------------------|----------|-------------------------------------------------------------|

> | `key`        | String | None             | yes      | The ssh public key to be added (name property)              |
> | `user`       | String | None             | yes      | The user who owns the `authorized_keys` file at `dir_path`  |

Reviewer: Name property should be in the Defaults column:
If there is no default, leave it blank:
```suggestion
| `key`        | String | Resource name    | yes      | The ssh public key to be added              |
| `user`       | String |                  | yes      | The user who owns the `authorized_keys` file at `dir_path`  |
```

| `group`      | String | `user`           | no       | The group who owns the `authorized_keys` file at `dir_path` |

> | `dir_path`   | String | `/home/#{user}'` | no       | Path to the directory containing the `authorized_keys` file |

Reviewer: Should this be `/home/user/.ssh`?

> | `group`      | String | `user`               | no       | The group who owns the `authorized_keys` file at `dir_path` |
> | `dir_path`   | String | `/home/#{user}/.ssh` | no       | Path to the directory containing the `authorized_keys` file |

Reviewer: Please change `No` entries in all Required columns to blank. 
```suggestion
| `group`      | String | `user`               |          | The group who owns the `authorized_keys` file at `dir_path` |
| `dir_path`   | String | `/home/#{user}/.ssh` |          | Path to the directory containing the `authorized_keys` file |
```
